src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1255,17 +1255,16 @@ private BoundStatement ChainImplicitStructConstructor(MethodSymbol methodSymbol,
                 var codeGen = new CodeGen.CodeGenerator(method, block, builder, moduleBuilder, diagnosticsForThisMethod, optimizations, generateDebugInfo);
 
                 // We need to save additional debugging information for MoveNext of an async state machine.
-                var stateMachineMethod = method as SynthesizedStateMachineMethod;
-                bool isStateMachineMoveNextMethod = stateMachineMethod != null && method.Name == WellKnownMemberNames.MoveNextMethodName;
+                var stateMachineMoveNextMethod = method as SynthesizedStateMachineMoveNextMethod;
```

Comment:
This change makes the code more fragile (tight coupling, by making the code sensitive to irrelevant refactoring of implementation types). If you want a "more precise" test than was there before, please add a virtual boolean property to SynthesizedStateMachineMethod or some supertype and test that property.